File: tasks/35_Search_Insert_Position.go
package main

func searchInsert(nums []int, target int) int {
	start := 0
	end := len(nums)

	for start < end {
		mid := (start + end) / 2
		cur := nums[mid]

		if cur == target {
			return mid
		} else if target > cur {
			start = mid + 1
		} else {
			end = mid
		}

	}

	return start
}

File: tasks/35_Search_Insert_Position.py
# https://leetcode.com/problems/search-insert-position/
from typing import List


class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        low = 0
        high = len(nums) - 1

        while low <= high:
            mid = (low + high) >> 1

            if nums[mid] == target:
                return mid

            if nums[mid] < target:
                low = mid + 1
            else:
                high = mid - 1

        return low
